### PR TITLE
JuniperIDP: data connector, parser, samples

### PR DESCRIPTION
**Pull request by averbn**
## Proposed Changes

  - add new data connector

### CODE REVIEW

**Review by v-ampami — CHANGES_REQUESTED** [review]

File: Solutions/JuniperIDP/Parsers/Juniper_IDP.txt [hunk 1]
```diff
@@ -0,0 +1,93 @@
+// Usage Instruction :
```
Comment (v-jayakal) [inline Solutions/JuniperIDP/Parsers/Juniper_IDP.txt:1]: Remove the all the column_ifexists, which are not necessary. Use project-rename when no transformation is needed.

**Review by v-jayakal — CHANGES_REQUESTED** [review]

File: Solutions/JuniperIDP/Data Connectors/Connector_LogAnalytics_agent_JuniperIDP.json [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    "connectivityCriterias": [
+        {
+            "type": "IsConnectedQuery",
+            "value": [
+                "JuniperIDP\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)"
+            ]
+        }
+    ],
+    "availability": {
+        "status": 1
```
[the remark this answers is not in the document]
Comment (v-jayakal) [reply Solutions/JuniperIDP/Data Connectors/Connector_LogAnalytics_agent_JuniperIDP.json:35]: Can you please add "isPreview": true here?

**Review by v-jayakal — CHANGES_REQUESTED** [review]

File: Solutions/JuniperIDP/Data Connectors/Connector_LogAnalytics_agent_JuniperIDP.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (averbn) [reply Solutions/JuniperIDP/Data Connectors/Connector_LogAnalytics_agent_JuniperIDP.json]: done

File: Solutions/JuniperIDP/Parsers/Juniper_IDP.txt (hunk 1 above)
Comment (averbn) [reply Solutions/JuniperIDP/Parsers/Juniper_IDP.txt:1]: Done

File: Solutions/JuniperIDP/Data Connectors/Connector_LogAnalytics_agent_JuniperIDP.json [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
...       "description":"Follow the configuration steps below to get Juniper IDP logs into Azure Sentinel. This configuration enriches events generated by Juniper IDP module to provide visibility on log source information for Juniper IDP logs. Refer to the [Azure Monitor Documentation](https://docs.microsoft.com/azure/azure-monitor/agents/data-sources-json) for more details on these steps.\n1. Download config file [juniper_idp.conf](link to the file).\n2. Login to the server where you have installed Azure Log Analytics agent.\n3. Copy juniper_idp.conf to the /etc/opt/microsoft/omsagent/**workspace_id**/conf/omsagent.d/ folder.\n4. Edit juniper_idp.conf as follows:\n\n\t i. change the listen port for receiving logs based on your configuration (line 3)\n\n\t ii. replace **workspace_id** with real value of your Workspace ID (lines 58,59,60,63)\n5. Save changes and restart the Azure Log Analytics agent for Linux service with the following command:\n\t\tsudo /opt/microsoft/omsagent/bin/service_control restart\n6. To configure a remote syslog destination, please reference the [SRX Getting Started - Configure System Logging](https://kb.juniper.net/InfoCenter/index?page=content&id=kb16502).",
```
Comment (v-rucdu) [inline Solutions/JuniperIDP/Data Connectors/Connector_LogAnalytics_agent_JuniperIDP.json:151]: Please add link for the conf file.

**Review by v-rucdu — CHANGES_REQUESTED** [review]

File: Solutions/JuniperIDP/Data Connectors/Connector_LogAnalytics_agent_JuniperIDP.json [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+                        "fillWith": [
+                            "WorkspaceId"
+                        ],
+                        "label": "Workspace ID"
+                    },
+                    "type": "CopyableLabel"
+                }
+            ]
+    }
+]
```
Comment (v-rucdu) [inline Solutions/JuniperIDP/Data Connectors/Connector_LogAnalytics_agent_JuniperIDP.json:164]: Please add a Metadata object within this file. See https://raw.githubusercontent.com/Azure/Azure-Sentinel/29c401e232cc7ce1ee91e7583133d3aa53e7e4af/DataConnectors/Templates/Connector_REST_API_template.json for further detail.

**Review by v-rucdu — APPROVED** [review]

**Review by v-jayakal — APPROVED** [review]

**Pull request merged**